### PR TITLE
Updates the deployment config files and adds Deployment doc for the `api` service

### PR DESCRIPTION
**Pull request by PuneetPunamiya**
# Changes

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [ ] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if functionality changed/added)
- [ ] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/pipeline/blob/master/CONTRIBUTING.md) for more details._

### CODE REVIEW

File: docs/DEPLOYMENT.md [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+
+### CRC
+
+### ko
+
+## Deploy Application on CRC
+
+### Deploy API Service
+
+Ensure you are in `cd backend/api` directory
```
Comment (pratap0007) [inline docs/DEPLOYMENT.md:13]: it should be `cd api`

File: docs/DEPLOYMENT.md [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+#### Deploy the database
+
+Export `KO_DOCKER_REPO` for ko to publish image to. E.g.
+
+```
+export KO_DOCKER_REPO=quay.io/<username>
+```
+
+```
+cd backend/api
```
Comment (pratap0007) [inline docs/DEPLOYMENT.md:24]: `cd api`

File: docs/DEPLOYMENT.md [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+
+NAME                  READY   STATUS                       RESTARTS   AGE
+db-5b9c486f65-5nxbc   1/1     Running                      0          2m59s
+db-migration-wqpwq    0/1     CreateContainerConfigError   0          18s
+                              ^^^^^^^^^^^^^^^^^^^^^^^^^^
+```
+
+#### Update the GitHub Api secret and client id
+
+Edit `api.yaml` and update the secret - `api`. Set GitHub `oauth` client id and client secret.
```
Comment (pratap0007) [inline docs/DEPLOYMENT.md:65]: I think don't have `api.yaml`  file

File: docs/DEPLOYMENT.md [hunk 4]
```diff
@@ -0,0 +1,132 @@
+# Development
```
Comment (SM43) [inline docs/DEPLOYMENT.md:1]: Deployment

File: docs/DEPLOYMENT.md [hunk 5]
```diff
@@ -0,0 +1,132 @@
...
+
+## Setup Tools
+
+### CRC
```
Comment (SM43) [inline docs/DEPLOYMENT.md:5]: could you please avoid `crc` `oc` and use `kubectl`
and also test in on gcp or kind..

Comment (PuneetPunamiya) [reply docs/DEPLOYMENT.md]: Works fine on Minikube

**Review by sthaha — COMMENTED** [review]
/lgtm
/approve
/meow

**Pull request merged**